packages/origin-247-certificate/src/offchain-certificate/offchain-certificate.service.ts

```diff
@@ -55,18 +57,33 @@ export class OffchainCertificateService<T = null> {
     public async claim(command: IClaimCommand): Promise<void> {
         const savedCommand = await this.certCommandRepo.save({ payload: command });
         const event = new CertificateClaimedEvent(command.certificateId, command);
-        const aggregate = await this.createAggregate(event);
+        const aggregate = await this.createAggregate([event]);
         await this.propagate(event, savedCommand, aggregate);
     }
 
     public async transfer(command: ITransferCommand): Promise<void> {
         const savedCommand = await this.certCommandRepo.save({ payload: command });
         const event = new CertificateTransferredEvent(command.certificateId, command);
-        const aggregate = await this.createAggregate(event);
+        const aggregate = await this.createAggregate([event]);
         await this.propagate(event, savedCommand, aggregate);
     }
 
     public async batchIssue(originalCertificates: IIssueCommandParams<T>[]): Promise<number[]> {
+        originalCertificates.forEach((params) => {
```

Comment:
Nie rozumiem co ten kod robi. To jest walidacja? Jeśli tak to zrobiłym to jakos bardziej explicit. To samo w tym `batchClaim`/transfer. Serio, dziwny kodzik xd

Reply:
Yup, this is the validation, which needs to happen on aggregate. I agree that it's not the cleanest. I've extracted them to separate functions so that it's more readable.